src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirOperationDataStore.cs

```diff
@@ -71,21 +72,93 @@ public async Task<ExportJobOutcome> CreateExportJobAsync(ExportJobRecord jobReco
+                        throw new JobNotFoundException(string.Format(Core.Resources.JobNotFound, id));
+                    }
+
+                    (string rawJobRecord, byte[] rowVersion) = sqlDataReader.ReadRow(V1.ExportJob.RawJobRecord, V1.ExportJob.JobVersion);
+
+                    return CreateExportJobOutcome(rawJobRecord, rowVersion);
+                }
+            }
         }
 
-        public Task<ExportJobOutcome> GetExportJobByHashAsync(string hash, CancellationToken cancellationToken)
+        public async Task<ExportJobOutcome> GetExportJobByHashAsync(string hash, CancellationToken cancellationToken)
         {
-            // TODO: Implement this method (returning null for now to allow the create method to run).
-            return Task.FromResult<ExportJobOutcome>(null);
+            EnsureArg.IsNotNullOrWhiteSpace(hash, nameof(hash));
+
+            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
+            using (SqlCommand sqlCommand = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                V1.GetExportJobByHash.PopulateCommand(sqlCommand, hash);
+
+                using (SqlDataReader sqlDataReader = await sqlCommand.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                {
+                    if (!sqlDataReader.Read())
+                    {
+                        return null;
+                    }
+
+                    (string rawJobRecord, byte[] rowVersion) = sqlDataReader.ReadRow(V1.ExportJob.RawJobRecord, V1.ExportJob.JobVersion);
+
+                    return CreateExportJobOutcome(rawJobRecord, rowVersion);
+                }
+            }
         }
 
-        public Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobRecord, WeakETag eTag, CancellationToken cancellationToken)
+        public async Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobRecord, WeakETag eTag, CancellationToken cancellationToken)
         {
-            // TODO: Implement this method.
-            return Task.FromResult(new ExportJobOutcome(jobRecord, eTag));
+            EnsureArg.IsNotNull(jobRecord, nameof(jobRecord));
+
+            byte[] rowVersionAsBytes = GetRowVersionAsBytes(eTag);
+
+            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
+            using (SqlCommand sqlCommand = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                V1.UpdateExportJob.PopulateCommand(
+                    sqlCommand,
+                    jobRecord.Id,
+                    jobRecord.Status.ToString(),
+                    jobRecord.QueuedTime,
+                    JsonConvert.SerializeObject(jobRecord, _jsonSerializerSettings),
+                    rowVersionAsBytes);
+
+                try
+                {
+                    var rowVersion = (int?)await sqlCommand.ExecuteScalarAsync(cancellationToken);
```

Comment:
Just curious why you are returning the rowVersion as an int here instead of the byte[] ?

Reply:
It seemed easier to do the conversion from bytes to an integer value in SQL, since the information is captured and stored as a string representation of a decimal in the `ExportJobOutcome.WeakETag` property on the C# side.

I am not tied to doing it this way though! Let me know if you have a nicer/cleaner implementation in mind.

Reply:
As discussed, I moved the conversion to the C# side and made use of the existing conversion methods for code readability in b6bad92.